datasets/plugins/batch/batch_dataset_plugin.py

```diff
@@ -0,0 +1,173 @@
+            run_id=run_id,
+            mode=mode,
+            class_field_name=class_field_name,
+        )
+
+    def _get_path_filters_columns(
+        self, columns: Optional[Union[Iterable[str], str]] = None, run_id: Optional[str] = None
+    ) -> Tuple[str, Optional[list], Optional[Iterable[str]]]:
+        path = self._get_dataset_path()
+        read_columns = self._get_read_columns(columns)
+        filters = None
+        query_run_id = run_id if run_id else self.run_id
+        if query_run_id:
+            filters = [("run_id", "=", query_run_id)]
+        return path, filters, read_columns
+
+    def read_pandas(
+        self,
+        columns: Optional[str] = None,
+        storage_format: str = "parquet",
+        run_id: Optional[str] = None,
+        **kwargs,
+    ) -> pd.DataFrame:
+        path, filters, read_columns = self._get_path_filters_columns(columns, run_id=run_id)
+
+        df: pd.DataFrame
+        if storage_format == "parquet":
+            df = pandas.read_parquet(path, columns=read_columns, engine="pyarrow", filters=filters, **kwargs)
+        elif storage_format == "csv":
+            df = pandas.read_csv(path, columns=read_columns, filters=filters, **kwargs)
+
+        for meta_column in self._META_COLUMNS:
+            if meta_column in df and (read_columns is None or meta_column not in read_columns):
+                del df[meta_column]
+        return df
+
+    def write(self, data: pd.DataFrame, **kwargs):
+        if not (self.mode & Mode.WRITE):
+            raise InvalidOperationException(f"Cannot write because mode={self.mode}")
+
+        if not isinstance(data, pd.DataFrame):
+            assert ValueError("data is not a pandas DataFrame")
+
+        if self.partition_by:
+            partition_cols = self.partition_by.split(",")
+        else:
+            partition_cols = list()
+
+        if self.path is None or "run_id" in partition_cols:
+            # Only partition on run_id if @dataset(path="s3://..") is not given
+            # or run_id is in partition_cols
+            if "run_id" not in partition_cols:
+                partition_cols.append("run_id")
+            self.run_id = self._executor.current_run_id  # DO NOT ALLOW OVERWRITE OF ANOTHER RUN ID
+            data["run_id"] = self.run_id
+
+        data.to_parquet(
+            self._get_dataset_path(),
+            engine="pyarrow",
+            compression="snappy",
+            index=False,
+            partition_cols=partition_cols,
+            **kwargs,
+        )
+
+    def read_dask(
+        self, columns: Optional[str] = None, run_id: Optional[str] = None, **kwargs
+    ) -> "dd.DataFrame":
+        import dask.dataframe as dd
+
+        path, filters, read_columns = self._get_path_filters_columns(columns, run_id=run_id)
+        return dd.read_parquet(
+            path,
+            columns=read_columns,
+            filters=filters if filters and len(filters) else None,
+            engine="pyarrow",
```

Comment:
I guess more broadly is there a way to set the engine for the plugin 😅  Sorry for the duplicate comments, just want to make sure we capture any change to this behavior consistently!